crates/gitbutler-hunk-dependency/src/stack.rs

```diff
@@ -47,4 +48,19 @@ impl StackRanges {
         }
         vec![]
     }
+
+    pub fn get_commit_dependencies(&self) -> HashMap<git2::Oid, HashSet<git2::Oid>> {
+        let data = self
+            .paths
+            .iter()
+            .flat_map(|(_, path)| path.commits.iter())
+            .collect_vec();
+        data.into_iter()
+            .fold(HashMap::new(), |mut acc, (commit_id, dependencies)| {
+                acc.entry(*commit_id)
+                    .and_modify(|existing_dependencies| existing_dependencies.extend(dependencies))
+                    .or_insert(dependencies.clone());
+                acc
+            })
```

Comment:
Just noticed, we declared the data variable for debugging, so we can drop that, skip the `.collect_vec()` and just return the concatenated the iter steps.